services/examples/import_netcdf.rs

```diff
@@ -0,0 +1,267 @@
+    },
+    test_data,
+    util::{config::get_config_element, tests::initialize_debugging_in_test},
+};
+use log::debug;
+use snafu::{OptionExt, ResultExt};
+use std::fs;
+
+#[derive(Debug, Clone)]
+struct NetCdfGroup {
+    name: String,
+    groups: Vec<NetCdfGroup>,
+    arrays: Vec<String>,
+}
+
+#[derive(Debug, Clone)]
+struct ConversionMetadata {
+    pub dataset_in: String,
+    pub dataset_out: PathBuf,
+}
+
+impl NetCdfGroup {
+    fn flatten(&self) -> Vec<Vec<String>> {
+        let mut out_paths = Vec::new();
+
+        for group in &self.groups {
+            out_paths.extend(group.flatten_mut(Vec::new()));
+        }
+
+        for array in &self.arrays {
+            out_paths.push(vec![array.clone()]);
+        }
+
+        out_paths
+    }
+
+    fn flatten_mut(&self, mut path_vec: Vec<String>) -> Vec<Vec<String>> {
+        let mut out_paths = Vec::new();
+
+        path_vec.push(self.name.clone());
+
+        for group in &self.groups {
+            out_paths.extend(group.flatten_mut(path_vec.clone()));
+        }
+
+        for array in &self.arrays {
+            let mut path_vec = path_vec.clone();
+            path_vec.push(array.clone());
+            out_paths.push(path_vec);
+        }
+
+        out_paths
+    }
+
+    fn conversion_metadata(
+        &self,
+        file_path: &Path,
+        out_root_path: &Path,
+    ) -> Vec<ConversionMetadata> {
+        let in_path = file_path.to_string_lossy();
+        let mut metadata = Vec::new();
+
+        for data_path in self.flatten() {
+            let md_path = data_path.join("/");
+            metadata.push(ConversionMetadata {
+                dataset_in: format!("NETCDF:\"{in_path}\":/{md_path}"),
+                dataset_out: out_root_path.join(md_path).with_extension("tiff"),
+            });
+        }
+
+        metadata
+    }
+}
+
+trait NetCdfVisitor {
+    fn group_tree(&self) -> Result<NetCdfGroup, NetCdfCf4DProviderError>;
+}
+
+impl NetCdfVisitor for MdGroup<'_> {
+    fn group_tree(&self) -> Result<NetCdfGroup, NetCdfCf4DProviderError> {
+        let mut groups = Vec::new();
+        for subgroup_name in self.group_names() {
+            let subgroup = self.open_group(&subgroup_name)?;
+            groups.push(subgroup.group_tree()?);
+        }
+
+        let dimension_names: HashSet<String> = self
+            .dimension_names()
+            .map_err(|source| NetCdfCf4DProviderError::CannotReadDimensions { source })?
+            .into_iter()
+            .collect();
+
+        let mut arrays = Vec::new();
+        for array_name in self.array_names() {
+            // filter out arrays that are actually dimensions
+            if dimension_names.contains(&array_name) {
+                continue;
+            }
+
+            arrays.push(array_name.to_string());
+        }
+
+        Ok(NetCdfGroup {
+            name: self.name.clone(),
+            groups,
+            arrays,
+        })
+    }
+}
+
+fn main() {
```

Comment:
is this example the importer? shouldn't this be under `bin` and accept parameters?

Reply:
otherwise just delete it. it'S just code that's redundant and will have to be maintained or gets out of data

Reply:
[a9e9190](https://github.com/geo-engine/geoengine/pull/526/commits/a9e9190b16e49b76daeb76b23fb6c0a7577639aa)